pkg/cmd/roachtest/cluster.go

```diff
@@ -1585,6 +1585,59 @@ func (c *clusterImpl) FetchCores(ctx context.Context, l *logger.Logger) error {
 	})
 }
 
+// FetchPebbleCheckpoints fetches any Pebble checkpoints on the cluster. These
+// are typically generated by failed consistency checks, i.e. replica
+// divergence, and only contain data for the inconsistent ranges.
+func (c *clusterImpl) FetchPebbleCheckpoints(ctx context.Context, l *logger.Logger) error {
+	// Don't grab checkpoints on empty clusters or local runs.
+	if c.spec.NodeCount == 0 || c.IsLocal() {
+		return nil
+	}
+
+	// Checkpoints can be large. Bail out after 3 minutes.
+	return timeutil.RunWithTimeout(ctx, "checkpoints", 3*time.Minute, func(ctx context.Context) error {
+		numStores := 1
+		if ssds := c.spec.SSDs; ssds > 1 {
+			numStores = ssds
+		}
+		for storeIdx := 1; storeIdx <= numStores; storeIdx++ {
+			// Find any checkpoints.
+			//
+			// TODO(erikgrinaker): expose and use SyncedCluster.NodeDir(). We can't
+			// use the {store-dir} expansion which only refers to the first store.
+			storePath := fmt.Sprintf("/mnt/data%d/cockroach", storeIdx)
+			var checkpointNodes option.NodeListOption
+			results, err := c.RunWithDetails(ctx, l, c.All(), "sudo", "/bin/bash", "-c",
+				fmt.Sprintf("'find %s/auxiliary -maxdepth 1 -type d -name checkpoints'", storePath))
+			if err != nil {
+				return err
+			}
+			for _, result := range results {
+				if result.Err != nil {
+					return result.Err
+				}
+				if output := strings.TrimSpace(result.Output(false /* decorate */)); output != "" {
+					l.Printf("found Pebble checkpoints in n%d:%s", result.Node, storePath)
+					checkpointNodes = append(checkpointNodes, int(result.Node))
+				}
+			}
+
+			// Fetch checkpoints.
+			if len(checkpointNodes) > 0 {
+				src := filepath.Join(storePath, "auxiliary", "checkpoints")
+				dest := filepath.Join(c.t.ArtifactsDir(), "checkpoints", fmt.Sprintf("data%d", storeIdx))
```

Comment:
Can `storeIdx` dup across nodes? Consider putting the node ID into the path too, e.g. like `"n%ds%d"`.

Reply:
>Can storeIdx dup across nodes?

Yes.

`Get()` will handle creating directories for different nodes when fetching from multiple nodes. I.e. if we're fetching ´data1` from node 3,5,7, it will create the directories `3.data1`, `5.data1`, and `7.data1`.